site/app/controllers/admin/PlagiarismController.php

```diff
@@ -345,15 +345,11 @@ public function saveNewPlagiarismConfiguration($new_or_edit, $gradeable_id = nul
             $old_array = json_decode($old_config, true);
             $regex_in_old = in_array("regex", $old_array);
             $regex_in_new = in_array("regex", $json_data);
```

Comment:
Unrelated to this change, but I think this is a bug here as `in_array` will always return false for both of these since these not flat arrays, but rather dictionaries. We should maybe be using `isset` to test for the existence of a key in the dictionary.

Investigating:
```php
<?php

$a = [
  'regex' => 'foo',
];

var_dump(in_array('regex', $a));  # prints bool(false)
var_dump(isset($a['regex']));     # prints bool(true)

$b = ['regex'];

var_dump(in_array('regex', $b));  # prints bool(true)
var_dump(isset($b['regex']));     # prints bool(false)
```

Reply:
Yep, good catch.  I have pushed a new change.